### PR TITLE
FEATURE: Improve formatting for Slack transcript messages

### PR DESCRIPTION
**Pull request by davidtaylorhq**
- Fix multi-line code blocks
- Add strikethrough support
- Fix HTML entities inside code blocks
- Do not process formatting inside code blocks
- Ensure links are never created with no URL
- Replace - with _ in emoji names

### CODE REVIEW

File: lib/discourse_chat/provider/slack/slack_message.rb [hunk 1]
```diff
@@ -29,11 +29,27 @@ def url
     def text
       text = @raw['text'].nil? ? "" : @raw['text']
 
+      pre = {}
+
+      # Extract code blocks and replace with placeholder
+      text = text.gsub(/```(.*)```/m) do |match|
```
Comment (ZogStriP) [inline lib/discourse_chat/provider/slack/slack_message.rb]: You might want to be lazy, just in case there are more than one code block? (is that even possible?)

```suggestion
      text = text.gsub(/```(.*?)```/m) do |match|
```

File: lib/discourse_chat/provider/slack/slack_message.rb [hunk 2]
```diff
@@ -46,14 +62,35 @@ def text
           next "@#{user_name}"
         end
 
-        "[#{text}](#{link})"
+        if link.nil?
+          text
+        elsif link == text
```
Comment (ZogStriP) [inline lib/discourse_chat/provider/slack/slack_message.rb:67]: ❤️

File: lib/discourse_chat/provider/slack/slack_message.rb [hunk 3]
```diff
@@ -46,14 +62,35 @@ def text
...
+          "<#{link}>"
+        else
+          "[#{text}](#{link})"
+        end
       end
 
       # Add an extra * to each side for bold
-      text = text.gsub(/\*(.*?)\*/) do |match|
+      text = text.gsub(/\*.*\*/) do |match|
```
Comment (ZogStriP) [inline lib/discourse_chat/provider/slack/slack_message.rb]: Why did you remove the lazy modifier?

File: lib/discourse_chat/provider/slack/slack_message.rb [hunk 4]
```diff
@@ -46,14 +62,35 @@ def text
...
         "*#{match}*"
       end
 
+      # Add an extra ~ to each side for strikethrough
+      text = text.gsub(/~.*?~/) do |match|
+        "~#{match}~"
+      end
+
+      # Replace emoji - with _
+      text = text.gsub(/:[a-z0-9_-]+:/) do |match|
```
Comment (ZogStriP) [inline lib/discourse_chat/provider/slack/slack_message.rb:85]: Shorter?

```suggestion
      text = text.gsub(/:[\w-]+:/) do |match|
```

**Review by ZogStriP — APPROVED** [review]

File: lib/discourse_chat/provider/slack/slack_message.rb (hunk 4 above)
Comment (davidtaylorhq) [reply lib/discourse_chat/provider/slack/slack_message.rb:85]: Slack emojis must be all-lowercase, and I think `\w` would include uppercase characters
